src/PlaneWaveBasis.jl

```diff
@@ -322,23 +340,26 @@ Creates a new basis identical to `basis`, but with a custom set of kpoints
     PlaneWaveBasis(basis.model, basis.Ecut,
                    basis.fft_size, basis.variational,
                    kcoords, kweights, kgrid, kshift,
-                   basis.symmetries_respect_rgrid, basis.comm_kpts)
+                   basis.symmetries_respect_rgrid, basis.comm_kpts, basis.architecture)
 end
 
 """
-    G_vectors(fft_size::Tuple)
+    G_vectors([architecture=AbstractArchitecture], fft_size::Tuple)
 
 The wave vectors `G` in reduced (integer) coordinates for a cubic basis set
 of given sizes.
 """
-function G_vectors(fft_size::Union{Tuple,AbstractVector})
+G_vectors(fft_size::Union{Tuple,AbstractVector}) = G_vectors(CPU(), fft_size)
+function G_vectors(architecture::AbstractArchitecture, fft_size::Union{Tuple,AbstractVector})
     # Note that a collect(G_vectors_generator(fft_size)) is 100-fold slower
     # than this implementation, hence the code duplication.
     start = .- cld.(fft_size .- 1, 2)
     stop  = fld.(fft_size .- 1, 2)
     axes  = [[collect(0:stop[i]); collect(start[i]:-1)] for i in 1:3]
-    [Vec3{Int}(i, j, k) for i in axes[1], j in axes[2], k in axes[3]]
+    Gs = [Vec3{Int}(i, j, k) for i in axes[1], j in axes[2], k in axes[3]]
+    to_device(architecture, Gs)
```

Comment:
Again unless you see any major issues with this I would make the calling party responsable to do the transfer.